MyFirstSlideshow/app/src/main/AndroidManifest.xml

```diff
@@ -2,6 +2,12 @@
 <manifest package="com.charles.myfirstslideshow"
     xmlns:android="http://schemas.android.com/apk/res/android">
 
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
```

Comment:
AndroidManifest.xml 6 to 9 : No point of keeping this state we just need to check the internet connectivity.Because all the things happen by user event